src/VisualStudio/TestUtilities/Workspace/Solution.cs

```diff
@@ -45,13 +45,27 @@ public string FileName
             }
         }
 
+        public void Save()
+        {
+            Directory.CreateDirectory(Path.GetDirectoryName(FileName));
+            DteSolution.SaveAs(FileName);
+        }
+
+        public Project OpenProject(string projectFilePath)
+        {
+            var dteProject = _dteSolution.AddFromFile(projectFilePath);
+            Save();
+            return new Project(dteProject, this, ProjectLanguage.CSharp);
+        }
+
         public Project AddProject(string projectName, ProjectTemplate projectTemplate, ProjectLanguage projectLanguage)
         {
             var solutionFolder = Path.GetDirectoryName(FileName);
-            var projectPath = Path.Combine(solutionFolder, projectName);
+            var projectPath = Path.Combine(solutionFolder, Path.GetFileNameWithoutExtension(projectName));
```

Comment:
According to DTE solution AddFromTemplate docs:

> ProjectName: Required. The name of the project file in the destination directory. This should include the extension. The name displayed is derived from ProjectName.

Unless I misunderstood I believe that this means that the projectName should also contain the csproj ending.